### DISCUSSION TITLE
Aidex issues in xdrip

### DISCUSSION THREAD

**Discussion by sjsbpl** [NightscoutFoundation/xDrip#3284] [General]
I am running an Aidex cgm using the xdrip hardware data source as Aidex app, but keep getting no readings on xdrip. When I do get readings it sometimes skips readings. 
![inbound6119857632143835687](https://github.com/NightscoutFoundation/xDrip/assets/132947573/235dc704-e289-426a-bb97-c050d5c23472)
![inbound5284464868647885465](https://github.com/NightscoutFoundation/xDrip/assets/132947573/8d86282f-7bd4-40af-b4e1-185032e7c578)
Any ideas ?

**Comment by Navid200** [comment]
Hi @andyrozman,

Does Aidex offer backfill? 

What could be causing this?

Thanks

**Comment by sjsbpl** [comment]
Update

Xdrip is still missing readings, it did all night whilst do not disturb was on so think that one issue but as that not on during the day that doesn't explain the daytime missed readings. My aim as I'm in the UK and am T2 and therefore pay for my sensors myself, I want to use a combination of Aidex and libre 2 to help lower the cost and use nightscout with both as I like the reports I can get with that. Unless anyone can come up with any other ideas.

**Reply by LWXJay** [reply]
I'm having the same issue with Aidex. It's fine with Dexcom, but I'm constantly missing readings (only ever one or two sequentially) when the BG's are passed from Aidex to xDrip+. They are all there when you check them in the Aidex app. This is preventing AAPS from doing it's job correctly which is impacting me. I'm keen to see this fixed also.

**Reply by Navid200** [reply]
Then, I have the same question for you.
What version of xDrip are you using?

**Comment by andyrozman** [comment]
Yes Aidex transmitter by itself offers backfill (app uses it all the time),
but we get only current values broadcasted.

Aidex app requires that Location and BT is always on, or else readings are
missed... So you have to have permission for Location and BT to be on
always. Problem is that for app to connect to sensor it needs to do do full
detection everytime, if location is off it can't do BT scan and reading is
not read.

It might also help setting Aidex App to run in foreground and permissions for Location and BT need to be on Always too. And Aidex App must NOT be killed in background...

Andy

**Reply by Navid200** [reply]
So, xDrip will not show readings if the user walks away for a while from the phone and then comes back even though the aidex app will.  Right?

**Reply by sjsbpl** [reply]
I'm not even away from the phone and xdrip will miss a reading that Aidex uses

**Reply by Navid200** [reply]
What is the phone brand and model?

**Reply by sjsbpl** [reply]
Google pixel 6a

**Reply by Navid200** [reply]
Have you set location service to be available to xDrip all the time?  Please see this:
https://navid200.github.io/xDrip/docs/Location.html

**Reply by sjsbpl** [reply]
Yes location is set to on

**Reply by Navid200** [reply]
That's not enough.
Did you see that link?

**Reply by sjsbpl** [reply]
Yes and I meant to say location is set to allow all the time on both xdrip and Aidex apps

**Reply by Navid200** [reply]
What version of xDrip are you using?

**Comment by andyrozman** [comment]
Location needs to be "on Always" for Aidex app too and Aidex App needs to be running in foreground (if your phone keeps killing it). Some of phones have very strong Power Management which keeps killing services in background.

@Navid200 Yes. Aidex App will get "backfill" values, but this values won't be broadcasted.

**Reply by Navid200** [reply]
I asked you a question to clarify what "broadcast" in this context means and you keep repeating the same word.
Will xDrip have missing points or not?

**Reply by andyrozman** [reply]
Yes Xdrip will receive only current value, when that value is received by Aidex App. Like I said, only current record is "broadcasted" to Xdrip (and AAPS). 
I don't know how can I explain it differently. Broadcast is Android message that one App send to specified another App. It is the same as Xdrip does to other applications (like AAPS).

**Reply by Navid200** [reply]
OK, got it.
I don't know all the technical details of every hardware data source until I make a change to it myself or use it as a user.  Without that, I didn't know aidex broadcasted to xDrip.   I remember you created a receiver for it.  So, I thought it may have behaved like Dexcom or Libre where there is nothing broadcasted.

So, from the user perspective, if the user walks away from the phone, he/she will end up with missing points in xDrip.
That's what a non-technical user will know and understand.  An aidex user may have no idea what broadcast means.

Thanks for helping me understand.  So, next time someone asks for support, I know what I am talking about.

**Reply by andyrozman** [reply]
Same can happen if Aidex App is killed (and therefore misses a reading). Aidex app and transmitter is very sensitive and also has weird way of connection to transmitter. Normal applications just take BT address and connects, but Aidex always needs to do BT scan (it can't connect directly with having address stored), and because it needs to do that Location needs to be on all the time.

**Comment by Navid200** [comment]
The two of you reporting issues with Aidex, we need to know:

1- How long have you been using it?

2- Have you always had this problem?

3- If not, when did it start?

4- If you go to a previous version of xDrip, does the problem continue?  You don't need to uninstall xDrip to install a different version.  You can just install the newer or older version over your current version.

5- If the problem only manifests itself when you use a particular version of xDrip, which version is it?

**Reply by LWXJay** [reply]
Hi Navid,

Thanks for your reply, much appreciated!

1. Two weeks for me, I've just changed from Dexcom to try out Aidex.
2. Yes
3. N/A
4. I'll try an earlier version and report back, are there any specific versions which you would recommend?
5. N/A

I'm currently using version 23ccbd1-2023.07.16
Code 2123071614

Please let me know what other information you need :).
-Jay

**Reply by Navid200** [reply]
What happens if you use this version?
https://github.com/NightscoutFoundation/xDrip/releases/tag/2024.01.05

**Reply by LWXJay** [reply]
Hey Navid, I've just installed that now and will update you in a few hours time :)

**Comment by LWXJay** [comment]
> What happens if you use this version? https://github.com/NightscoutFoundation/xDrip/releases/tag/2024.01.05

I'm sorry to report it's no better. I'm still dropping readings all over the place in xDrip+

**Reply by andyrozman** [reply]
Problem is not in Xdrip but in Aidex application (in further text AA). It is missing reading (which it later gets with Backfill, but those backfills are not "transmitted" forward to listening apps). If AA would be getting all readings, all readings would be in Xdrip. Did you set up permission for Location to Always (for AA)? Do you have it running in foreground? You need to prevent your phone to killing AA in background, try to set all Power management to off. Some phones have too "powerful" (and wrongly implemented) Power Mangament which kills everything. There is "Don't Kill UP my App" site, maybe they have some ideas what to do. Can you try it with older phone (just for test), maybe something that has still Android 12 on it? With Android 13/14 PM seems to get worse, even on same phone.

**Reply by LWXJay** [reply]
Okay, if the issue is Aidex, I guess we can't do anything to fix the issue. I have ensures all permissions and power management are correct. Depending on what I'm doing with my phone, I can loose many more readings, if playing Fornite or something very demanding I've noted. It looks like the application isn't getting the resources it needs and xDrip can miss many, many readings which would certainly support the fact that Aidex is getting backfill readings. I've not tried the "don't Kill Up my App", but I've also changed back to Dexcom from Aidex. It was a pretty disappointing experiencing using it anyway (as a CGM product). I don't have any non Android 13 phones (nothing older) so can't test with that either sorry.

**Comment by sjsbpl** [comment]
Have accepted that up to 3 readings an hour seem to be either not broadcast by Aidex app or being received by xdrip, but usually have do not disturb on at night and seem to not get any readings on xdrip for most of this time, even with override do not disturb set to on in Aidex app. Have today turned do not disturb off on phone so will see what happens tonight.

**Comment by kenkit** [comment]
This issue persists even in the iphone. If the phone locks or goes off, aidex will still get backfills but apple health will miss all those readings. But when it's active it works just well. It makes me wonder are they trying to keep people from using the app as a health data source ?
